homeassistant/components/backup/manager.py

```diff
@@ -325,10 +325,26 @@ async def async_get_backup(
 
         return (backup, agent_errors)
 
-    async def async_delete_backup(self, backup_id: str) -> None:
+    async def async_delete_backup(self, backup_id: str) -> dict[str, Exception]:
         """Delete a backup."""
-        for agent in self.backup_agents.values():
-            await agent.async_delete_backup(backup_id)
+        agent_errors: dict[str, Exception] = {}
+        agent_ids = list(self.backup_agents.keys())
```

Comment:
We can remove the same leftover `.keys()` on line 293.
```suggestion
        agent_ids = list(self.backup_agents)
```